bridge/bridgeapp/views.py

```diff
@@ -47,26 +52,38 @@ def get(self, request, category_id, category_slug):  #category_slug needed for U
     def post(self, request, category_id, category_slug):
         """Create new thread & select applicable categories- Many to many relationship between categories and threads"""
         form = ThreadForm(request.POST)
+        # assigns category_id parameter to id variable and category_slug parameter to slug variable
         id, slug = category_id, category_slug
+        # if statement tests whether the form is valid, and if not valid the form will not post
         if form.is_valid():
+            # assigns cleaned form data (via built-in Django method) to the data variable
             data = form.cleaned_data
+            # assigns the data entered in form by user to question and cat_ids variables
             question, cat_ids = data['body'], data['category_ids']
+            # tests whether user has selected at least 1 category and prevents form submission if not true
             if cat_ids:
+                # assigns newly created Thread object containing user-entered data to thread variable
                 thread = Thread.objects.create(body=question)
+                # loops through category ids selected by user and adds selected categories as foreign key to the current thread
                 for cat_id in cat_ids:
                     category = Category.objects.get(id=cat_id)
                     thread.categories.add(category)
+                # assigns the first item in the cat_ids list to id variable and assigns the slugified cat_ids type attribute to slug variable
                 id, slug = cat_ids[0], slugify(
                     Category.objects.get(id=cat_ids[0]).type)
-
-        return redirect('category', category_id=id, category_slug=slug)  #redirects to first selected Category
+        # redirects to view for first selected Category
```

Comment:
Redirect is to a url, not a view.

The 1st parameter for the `redirect()` method is the name 'category' for the url path `bridge/categories/<str:category_id>/<str:category_slug>`.

When the url dispatcher (`urls.py`) receives the redirect request it then send the request to the `BridgeCategoryView` for processing.